src/matching/Match_patterns.ml

```diff
@@ -365,9 +399,17 @@ let check2 ~hook mvar_context range_filter (config, equivs) rules
            * TODO: bloom filter was removed, undo this inlining?
            *)
           let visit_stmt () =
+            let inside_function = env.extra.inside_function in
```

Comment:
here is an example of where there is a bunch of introduced code which is not necessarily clearly related to the outer context